src/o365/teams/commands/deeplink/tab-generate.ts

```diff
@@ -0,0 +1,188 @@
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import GraphCommand from '../../../base/GraphCommand';
+import Utils from '../../../../Utils';
+import request from '../../../../request';
+import { Tab } from '../../Tab';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+export enum TabTypeOptions {
+  Static = "Static",
+  Configurable = "Configurable"
+}
+
+export interface Deeplink {
+  deeplink: string;
+}
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  teamId: string;
+  channelId: string;
+  tabId: string;
+  label: string;
+  tabType: string;
+}
+
+class TeamsDeeplinkTabGenerateCommand extends GraphCommand {
```

Comment:
We should implement the `getTelemetryProperties` method to provide command telemetry.